package.json

```diff
@@ -89,6 +89,58 @@
         "path": "./syntaxes/terraform.tmGrammar.json"
       }
     ],
+    "semanticTokenTypes": [
+      {"id": "hcl-attrName", "superType": "property"},
+      {"id": "hcl-blockType", "superType": "type"},
+      {"id": "hcl-blockLabel", "superType": "enumMember"},
+
+      {"id": "hcl-bool", "superType": "keyword"},
+	    {"id": "hcl-string", "superType": "string"},
+	    {"id": "hcl-number", "superType": "number"},
+	    {"id": "hcl-objectKey", "superType": "parameter"},
+	    {"id": "hcl-mapKey", "superType": "parameter"},
+	    {"id": "hcl-keyword", "superType": "variable"},
+	    {"id": "hcl-traversalStep", "superType": "variable"},
+	    {"id": "hcl-typeCapsule", "superType": "keyword"},
+	    {"id": "hcl-typePrimitive", "superType": "keyword"}
+    ],
+    "semanticTokenScopes": [
+      {
+        "scopes": {
+          "hcl-attrName": ["variable.other.property"],
+          "hcl-blockType": ["entity.name.type"],
+          "hcl-blockLabel": ["variable.other.enummember"],
+
+          "hcl-bool": ["keyword.control"],
+          "hcl-string": ["string"],
+          "hcl-number": ["constant.numeric"],
+          "hcl-objectKey": ["variable.parameter"],
+          "hcl-mapKey": ["variable.parameter"],
+          "hcl-keyword": ["keyword.control"],
+          "hcl-traversalStep": ["variable.other.readwrite"],
+          "hcl-typeCapsule": ["keyword.control"],
+          "hcl-typePrimitive": ["keyword.control"]
+        }
```

Comment:
The scopes here just follow VSCode's defaults per https://github.com/microsoft/vscode/blob/02e8bd149ca3fb68e7f26ebec33f920ba0feef96/src/vs/platform/theme/common/tokenClassificationRegistry.ts#L541-L582

Reply:
As per conversation in https://github.com/microsoft/vscode/issues/145312 we could make these more specific, e.g. by appending `.terraform` to all scopes, but I'm not sure what is the best long-term strategy for token/scope "parity", so I just mostly mimicked the existing behaviours for semantic tokens for now.